docs/best_practices.md

```diff
@@ -0,0 +1,82 @@
+# Programming Best Practices
+
+This guide is a list of things you can do to become a not only a better programmer, but to become a better programmer on a team.  When multiple people are contributing to the same code base there is a level of discipline required to make sure it's clean and consistent.
+
+Below are several things to keep in mind when designing and implementing your code. 
+
+## The Zen of Python, by Tim Peters
+
+- Beautiful is better than ugly.
+- Explicit is better than implicit.
+- Simple is better than complex.
+- Complex is better than complicated.
+- Flat is better than nested.
+- Sparse is better than dense.
+- Readability counts.
+- Special cases aren't special enough to break the rules.
+- Although practicality beats purity.
+- Errors should never pass silently.
+- Unless explicitly silenced.
+- In the face of ambiguity, refuse the temptation to guess.
+- There should be one - and preferably only one - obvious way to do it.
+- Although that way may not be obvious at first unless you're Dutch.
+- Now is better than never.
+- Although never is often better than *right* now.
+- If the implementation is hard to explain, it's a bad idea.
+- If the implementation is easy to explain, it may be a good idea.
+- Namespaces are one honking great idea - let's do more of those!
+
+## Should This Be Tested?
+
+- Data Collection (**Yes**)
+    - *Views*/APIs (**get** and **post** methods and Django class-based views excluded)
+    - *Forms* (**customized validation**, any customizations to Django base forms)
+- 3rd Party Tools (**No**)
+    - Django ORM
+    - Python Library
+- Algorithm (**Yes**)
+- Data Manipulation (**Yes**)
+
+## Should This Be Mocked?
```

Comment:
What is the term 'Mocked' used for?

Reply:
a Mock is a fake thing that acts like a real thing, used for testing. For example, when you want to test that your system can query an external API and manipulate the data it got back, but imagine you don't have an internet connection...you could mock the part of your application that actually does a request to the external system, and return data from a file instead.

There are mocking utilities built into python and Django to help this process. We'll get into this more when we do some unittests.